Series triple, transformation and processing updates (#603)
ietf-rats-wg/draft-ietf-rats-corim: draft-ietf-rats-corim.md

### Conditional Endorsement Series Triple {#sec-comid-triple-cond-series}

> The Conditional Endorsement Series Triple is used when an Endorser wants to add Claims with differing values based on conditions that have differing Claims values.

Reviewer: ```suggestion
The Conditional Endorsement Series Triple is used when an Endorser wants to match Claims of the same type (for example `version`) but with differing values.  If the conditions are met, new Claims are added as Endorsements.
```

> For example, if version 1.0 of a component has a Common Vulnerability and Exporsure (CVE), but version 2.0b does not, a series condition might contain `version` = "1.0" and an addition `cve` = "CVE_PLACEHOLDER". Another series condition might contain `version` = "2.0" and an addition `cve` = "NONE".

Reviewer: ```suggestion
For example, if version 1.0 of a component has a Common Vulnerability and Exposure (CVE), but version 2.0 does not, a series condition might contain `version` = "1.0" and an addition `cve` = "CVE_PLACEHOLDER". Another series condition might contain `version` = "2.0" and an addition `cve` = "CVE_NONE".
```

As soon as the condition value is matched the remaining conditions can be ignored.

> The Conditional Endorsement Series Triple has two forms of conditions, those that are in the series list and those that are common to all series items.

Reviewer: just improve Line 1421 and 1422

> The Conditional Endorsement Series Triple has two forms of conditions, those that are in the series list and those that are common to all series items.

Reviewer: ```suggestion
The Conditional Endorsement Series Triple has two forms of conditions,  those that are common to all series list items and those that are specific to a particular item in the series.
```

The `conditional-endorsement-series-triple-record` has the following parameters:

* `common-condition`: Matching conditions that are common to every item in the series.
Common conditions include `environment-map`, (optional) `claims-list`, and (optional) `authorized-by`.

> * `series`: A list of `conditional-series-record` items where each item's condition uses the same type measurements, but differ in value.

Reviewer: It is not clear with the line:

` each item's condition uses the same type measurements, but differ in value.`

> * `series`: A list of `conditional-series-record` items where each item's condition uses the same type measurements, but differ in value.

Reviewer: ```suggestion
* `series`: A list of `conditional-series-record` items where each item's condition uses the same type of measurements, meaning same label(code point) inside measurement-values-map, but differ in value.
```

~~~ pseudocode
FUNC transform(
    T: conditional-endorsement-series-triple-record,
    signer: [ + $crypto-key-type-choice ],
    profile: $profile-type-choice
) -> evs-item {
    evsitem := evs-item::NEW()
    FOREACH csr in T.series:
       // stage the condition
       sitem := series-item::NEW()
       sitem.condition.environment = T.condition.environment
       ems1 = mms_to_ems(T.condition.claims-list)
       sitem.condition.element-list = ems1
       ems2 := mms_to_ems(csr.condition)
       sitem.condition.element-list::APPEND(ems2)
       IF T.condition.authorized-by:
           sitem.condition.authority = T.condition.authorized-by
       ELSE_IF csr.series.condition.authorized-by:
           sitem.condition.authority = csr.series.condition.authorized-by
       // stage the addition
       sitem.addition.environment = T.condition.environment
       ems3 = mms_to_ems(csr.addition)
       sitem.addition.element-list = ems3
       sitem.addition.cmtype = `endorsements`
       sitem.addition.authority = signer
       IF profile:
           sitem.addition.profile = profile

>        evsitem[index-of(sitem)] = sitem.

Reviewer: ```suggestion
       evsitem[index-of(sitem)] = sitem
```
